app/controllers/declaration/AddPersonController.scala

```diff
@@ -66,10 +74,25 @@ class AddPersonController @Inject () (
     }
   }
 
-  private def isResponsiblePerson(data: AddPerson): Boolean = {
+  def updateFormErrors(f: InvalidForm, businessType: Option[BusinessType]): InvalidForm = {
+    val message = businessType match {
+      case Some(BusinessType.LimitedCompany) => Messages("error.required.declaration.add.position.for.limitedcompany")
+      case Some(BusinessType.SoleProprietor) => Messages("error.required.declaration.add.position.for.sole.proprietor")
+      case Some(BusinessType.Partnership) => Messages("error.required.declaration.add.position.for.partner.ship")
+      case Some(BusinessType.LPrLLP) => Messages("error.required.declaration.add.position.for.lprlpp")
+      case Some(BusinessType.UnincorporatedBody) => Messages("error.required.declaration.add.position.for.unicorporated.body")
+      case _ => throw new IllegalArgumentException("[Controllers][AddPersonController] business type is not known")
+    }
+
+    val newErrors: Seq[(Path, Seq[ValidationError])] = f.errors.map {
```

Comment:
f.errors.map { errors =>
    val s = if(errors._1 == Path("positions") { Seq(ValidationError(Seq(message)) } else { errors._2 }
   (errors._1, s)
}

Reply:
Possibly

Reply:
What do `p` and `s` mean? I am confused